lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -2135,7 +2139,15 @@ def cmd_unset(*args)
   #   stage since the command itself has been completed.
   def cmd_unset_tabs(str, words)
     datastore = active_module ? active_module.datastore : self.framework.datastore
-    datastore.keys
+    keys = datastore.keys
+
+    mod = active_module
+    mod.options.each do |name, mod_opt|
+      next if Msf::OptCondition.show_option(mod, mod_opt)
+      i = keys.index(name)
+      keys.delete_at i if i
+    end
```

Comment:
If there is no `active_module` this would be fooled into potentially removing keys from the framework datastore which is set on L2141.